src/frontend/js/project-service.js

```diff
@@ -56,18 +68,44 @@ angular.module('seashell-projects', ['seashell-websocket', 'marmoset-bindings'])
          * @constructor Constructor for a new SeashellFile
          * @param {SeashellProject} project - Project for which this file belongs in.
          * @param {String} name - Full name of file.
+         * @param {String} contents - Complete file contents.
          * @param {bool} is_dir - Is directory?
          * @param {Number} last_saved - Last saved time.
          */
-        function SeashellFile(project, name, is_dir, last_saved) {
+        function SeashellFile(project, name, contents, is_dir, last_saved) {
           var self = this;
           self.name = name.split("/");
+          var par = self.name.slice(0,self.name.length-1).join('/');
+          if(par.length>0) {
+            self.parent = project.root.find(par);
+          }
           self.project = project;
+          self.contents = contents ? contents : null;
```

Comment:
Why do we need contents?  With offline mode ready, shouldn't a call to read() be OK?